src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -480,15 +480,8 @@ void dispense_negative_deltas(
             schedule.at(k) += share;
             delta -= share;
         }
-        // the rest equally
-        auto share = std::div(delta, ss::smp::count);
-        for (quota_t& s : schedule) {
-            s += share.quot;
-            if (share.rem < 0) {
-                s += -1;
-                share.quot -= -1;
-            }
-        }
+        // distribure whatever is left in the delta equally between shards
+        dispense_equally(schedule, delta);
```

Comment:
the comment states this commit is a refactoring commit. but looking at `dispense_equally` i see that the `share.rem < 0` case:

```
        } else if (share.rem < 0) {
            v -= 1;
            share.quot += 1;
        }
```

looks like it is opposite the logic here that is being replaced, as well as handling the `share.rem > 0` case. so the commit message needs to explain what is changing, or why nothing has changed. most likely im missing something obvious, but at least a few minutes looking it doesn't seem to be equivalent.

Reply:
I have also fallen into this trap with this code. It was
```
s += -1;
```
and it became
```
v -= 1;
```
so no change in logic actually.

But anyway this code is gone now so that does not matter.

Reply:
oh, i totally missed that. thanks